recipe-client-addon/bootstrap.js

```diff
@@ -68,7 +70,13 @@ this.startup = function() {
   CleanupManager.addCleanupHandler(
     () => Preferences.ignore(PREF_LOGGING_LEVEL, LogManager.configure));
 
-  RecipeRunner.init();
+  // Initialize experiments first to avoid a race between initializing prefs
+  // and recipes rolling back pref changes when experiments end.
+  PreferenceExperiments.init().catch(err => {
+    log.error("Failed to initialize preference experiments:", err);
```

Comment:
Chatted about this in IRC - using the public `AddonManager` API won't be great for this since built-in system add-ons cannot be uninstalled or disabled, so re-running `bootstrap.js` `startup()` isn't really doable.

I'd suggest either simplifying`bootstrap.js` and moving the code you want to test into a different JSM that you write a test for, or if you want to have more control over the `AddonManager` lifecycle we have convenience methods for setting up and restarting AddonManager in http://searchfox.org/mozilla-central/source/toolkit/mozapps/extensions/internal/AddonTestUtils.jsm

Another option would be to use mochitest and restart the whole browser, but that's a bit more heavy-handed.